### PR TITLE
Demo kdf

### PR DESCRIPTION
**Pull request by paulidale**
Add demo code for PBKDF2 using SHA256 and SCRYPT.

Fixing #14107 and #14108.

Test vectors from RFC 7914.

### CODE REVIEW

File: demos/kdf/pbkdf2.c [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+#include <openssl/kdf.h>
+#include <openssl/obj_mac.h>
+#include <openssl/params.h>
+
+/*
+ * test vector from
+ * https://datatracker.ietf.org/doc/html/rfc7914
+ */
+
+static unsigned char password[] = {
```
Comment (slontis) [inline demos/kdf/pbkdf2.c:22]: Maybe a comment related to embedding passwords in production code is not good practice.

**Review by slontis — COMMENTED** [review]
There is a readme file in the demos folder that needs updating.

File: demos/kdf/pbkdf2.c (hunk 1 above)
Comment (paulidale) [reply demos/kdf/pbkdf2.c:26]: Good point, I'll add comments to the parameters/inputs.

**Review by slontis — APPROVED** [review]

**Comment by paulidale** [comment]
Merged, thanks for the review.

**Pull request closed without merging**